src/patch.js

```diff
@@ -19,6 +14,44 @@ export default class Patch {
     return new Iterator(this)
   }
 
+  rebalance () {
+    this.createBackbone()
+    this.createBalancedTree()
+  }
+
+  createBackbone () {
```

Comment:
Another presentation of the algorithm named this "vine" instead of "backbone" which I think is a bit nicer. Pretty subjective obviously but thought I would mention it.